Userland/Libraries/LibWeb/HTML/Scripting/TemporaryExecutionContext.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2023, Tim Flynn <trflynn89@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibJS/Forward.h>
+#include <LibWeb/Forward.h>
+
+namespace Web::HTML {
+
+// FIXME: When JS is run from outside the context of any user script, we currently do not have a running execution
+//        context. This results in a crash when we access VM::running_execution_context(). Until we figure out how to
+//        ensure we always have a running execution context, this is a workaround to temporarily push one.
+class TemporaryExecutionContext {
+public:
+    explicit TemporaryExecutionContext(EnvironmentSettingsObject&);
+    explicit TemporaryExecutionContext(DOM::Document&);
+    explicit TemporaryExecutionContext(JS::Realm&);
```

Comment:
Not a huge fan of providing all these overloads, in most cases we'll have easy access to all three and then noone knows which one to use. Can we settle on one and defer the indirection to the caller?

Reply:
:shrug: sure